lib/std/time.zig

```diff
@@ -344,6 +344,221 @@ test Timer {
+
+        pub const Options = struct {
+            offset: IMinutes = 0,
+        };
+
+        pub fn fromDaySeconds(seconds: DaySubseconds, options: Options) Self {
+            const fs_offset = @as(IDaySubseconds, options.offset) * subseconds_per_min + seconds;
+            var fs = std.math.comptimeMod(fs_offset, subseconds_per_day);
+
+            const hour: Hour = @intCast(@divFloor(fs, subseconds_per_hour));
+            fs -= @as(DaySubseconds, @intCast(hour)) * subseconds_per_hour;
+
+            const minute: Minute = @intCast(@divFloor(fs, subseconds_per_min));
+            fs -= @as(DaySubseconds, @intCast(minute)) * subseconds_per_min;
+
+            const second: Second = @intCast(@divFloor(fs, subseconds_per_s));
+            fs -= @as(DaySubseconds, @intCast(second)) * subseconds_per_s;
+
+            return .{
+                .hour = hour,
+                .minute = minute,
+                .second = second,
+                .subsecond = @intCast(fs),
+                .offset = options.offset,
+            };
+        }
+
+        pub fn toDaySeconds(self: Self) DaySubseconds {
+            return self.toDaySecondsZone(self.offset);
+        }
+
+        pub fn toDaySecondsZone(self: Self, zone: IMinutes) DaySubseconds {
+            var sec = @as(IDaySubseconds, zone) * 60 * multiplier;
+            sec += @as(IDaySubseconds, self.hour) * subseconds_per_hour;
+            sec += @as(IDaySubseconds, self.minute) * subseconds_per_min;
+            sec += @as(IDaySubseconds, self.second) * subseconds_per_s;
+            sec += @as(IDaySubseconds, self.subsecond);
+
+            return std.math.comptimeMod(sec, s_per_day * multiplier);
+        }
+
+        /// Does not handle leap seconds.
+        pub fn addWithOverflow(
+            self: Self,
+            hour: i64,
+            minute: i64,
+            second: i64,
+            subsecond: i64,
+        ) struct { time: Self, day_overflow: i64 } {
+            const fs = subsecond + self.subsecond;
+            const s = second + self.second + @divFloor(fs, 1000);
+            const m = minute + self.minute + @divFloor(s, 60);
+            const h = hour + self.hour + @divFloor(m, 60);
+            const overflow = @divFloor(h, 24);
+
+            return .{
+                .time = .{
+                    .subsecond = std.math.comptimeMod(fs, 1000),
+                    .second = std.math.comptimeMod(s, 60),
+                    .minute = std.math.comptimeMod(m, 60),
+                    .hour = std.math.comptimeMod(h, 24),
+                },
+                .day_overflow = overflow,
+            };
+        }
+
+        /// Does not handle leap seconds.
+        pub fn add(self: Self, hour: i64, minute: i64, second: i64, subsecond: i64) Self {
+            return self.addWithOverflow(hour, minute, second, subsecond).time;
+        }
+
+        pub fn fromRfc3339(str: []const u8) !Self {
+            if (str.len < "hh:mm:ssZ".len) return error.Parsing;
```

Comment:
[RFC3339](https://datatracker.ietf.org/doc/html/rfc3339) also allows numeric offsets from UTC as in '+08:00' for 8 hours ahead of UTC. If we only allow 'Z' i.e. UTC, this should be clearly documented.

Reply:
This is just to check that the minimum length is met. If you scroll down further you'll see both are allowed.